alea/model.py

```diff
@@ -230,28 +229,27 @@ class MinuitWrap:
             the dictionary passed to the Minuit call.
             """
 
-            def __init__(self, f, s_args):
+            def __init__(self, f, parameters: Parameters):
                 self.func = f
-                self.s_args = s_args
-                self.func_code = make_func_code(s_args)
+                self.s_args = parameters.names
+                self._parameters = {p.name: p.fit_limits for p in parameters}
 
             def __call__(self, *args):
                 return self.func(args)
 
         # Make the Minuit object
-        cost.errordef = Minuit.LIKELIHOOD
         m = Minuit(
-            MinuitWrap(cost, s_args=self.parameters.names),
+            MinuitWrap(cost, parameters=self.parameters),
             **defaults)
+        m.errordef = Minuit.LIKELIHOOD
         fixed_params = [] if fixed_parameters is None else fixed_parameters
```

Comment:
Not quite sure where, but this code that worked in main is not working now: 

```
from alea.examples import gaussian_model
simple_model = gaussian_model.GaussianModel()
simple_model.data = simple_model.generate_data(mu=0, sigma=2)
bf, ret = simple_model.fit(sigma=2, mu=2)
bf, ret
```

Reply:
Did it work now (since you marked it as resolved?)